tensorflow_addons/seq2seq/beam_search_decoder.py

```diff
@@ -180,10 +180,12 @@ def _check_static_batch_beam_maybe(shape, batch_size, beam_width):
     """Raises an exception if dimensions are known statically and can not be
     reshaped to [batch_size, beam_size, -1]."""
     reshaped_shape = tf.TensorShape([batch_size, beam_width, None])
-    if (batch_size is not None and shape.dims[0].value is not None
-            and (shape[0] != batch_size * beam_width or
-                 (shape.ndims >= 2 and shape.dims[1].value is not None and
-                  (shape[0] != batch_size or shape[1] != beam_width)))):
+    assert len(shape.dims) > 0
+    known_size = batch_size is not None and shape.dims[0].value is not None
+    case_1 = (shape[0] == batch_size * beam_width)
```

Comment:
Yeah, I was worried about that but since the code didn't catch that before and I wanted to keep the change minimal... I can add an explicit assertion (i.e. keep the failure) or just return False with the warning message here (and hope the rest of the code catches that / makes this never happen) -- I'd prefer the former, if that makes sense?